ingestionpolicies.go

```diff
@@ -103,7 +103,45 @@ func (p *IngestionPolicies) Create(policy *IngestionPolicyRequest) (*IngestionPo
 	return &ingestionPolicy, nil
 }
 
-func (p *IngestionPolicies) Update(policy *IngestionPolicyRequest) (*IngestionPolicyResponse, error) {
+func (p *IngestionPolicies) Update(policy *IngestionPolicyResponse) error {
+
+	var accounts []string
+	for _, v := range policy.Accounts {
+		accounts = append(accounts, v.ID)
+	}
+
+	var groups []string
+	for _, v := range policy.Groups {
+		groups = append(groups, v.ID)
+	}
+
+	var pointTags []map[string]string
+	for _, v := range policy.Tags {
+		pointTags = append(pointTags, map[string]string{"Key": v.Key, "Value": v.Value})
+	}
+
+	policyRequest := &IngestionPolicyRequest{
+		ID:          policy.ID,
+		Name:        policy.Name,
+		Description: policy.Description,
+		Scope:       policy.Scope,
+		Accounts:    accounts,
+		Groups:      groups,
+		Sources:     policy.Sources,
+		Namespaces:  policy.Namespaces,
+		Tags:        pointTags,
+	}
+
+	policy, err := p.RawUpdate(policyRequest)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (p *IngestionPolicies) RawUpdate(policy *IngestionPolicyRequest) (*IngestionPolicyResponse, error) {
```

Comment:
I think RawUpdate is fine because it is a method of IngestionPolicies. You are correct, Go does not allow method overloading, but Go will still let you have a RawUpdate method on multiple types in this package.  

Just make sure to rename this to `rawUpdate` with a lowercase r.  That way it doesn't get exported into the public API of your package.

Reply:
Oh interesting, lol, I hadn't even noticed the title case on the methods did that. 😆 Go does a lot of really subtle stuff that seem like you "just need know".

Part of me wasn't sure if I should leave this public, in case someone didn't have a response object. But I could also see it being confusing having two update methods.

Reply:
If they are updating, they probably will have a response object because they would have to fetch the original values before updating them.